File: app/src/main/kotlin/com/fishweeb/practical/EntityBase.java
package com.fishweeb.practical;

import android.graphics.Canvas;
import android.view.SurfaceView;

public interface EntityBase
{
    boolean IsDone();
    void SetIsDone(boolean _isDone);
    void Init(SurfaceView _view);
    void Update(float _dt);
    void Render(Canvas _canvas);

    int GetRenderLayer();
    void SetRenderLayer(int _newLayer);
}

File: app/src/main/java/edu/androidgroupproject/EntityBase.kt
package edu.androidgroupproject

import android.graphics.Canvas
import android.view.SurfaceView

interface EntityBase {
    fun IsDone(): Boolean
    fun SetIsDone(_isDone: Boolean)
    fun Init(_view: SurfaceView)
    fun Update(_dt: Float)
    fun Render(_canvas: Canvas)

    fun GetRenderLayer(): Int
    fun SetRenderLayer(_newLayer: Int)
}

